### ISSUE TITLE
outDir or declarationDir not working as expected

### ISSUE THREAD

**Issue by alexjoverm** [s-panferov/awesome-typescript-loader#388]
Hi,

I had https://github.com/alexjoverm/typescript-library-starter running on ts-loader. The types were generated on dist/types by the `outDir` setting. In ATL none `outDir` or `declarationDir` are working as expected. It's placing the types under `dist/src` (dist + the original source code dir).

It seems to be a bug. Probably relates to #247. I tried the workaround suggested there and no success.

**To reproduce:**

```shell
git clone https://github.com/alexjoverm/typescript-library-starter.git
cd typescript-library-starter

# Run npm install and write your library name when asked.
npm install
npm run build
```

Then you'll see that types are generated under dist/src instead of dist/types

**Comment by s-panferov** [comment]
@alexjoverm thanks for the example, landed into awesome-typescript-loader@3.1.0

**Comment by alexjoverm** [comment]
Thanks for the great work!
